### PR TITLE
change squeeze in gammatone process method

### PR DESCRIPTION
**Pull request by hoyer-a**
Adjusted `GammatoneBands` class to keep the cshape of the input signal and have it consistent with the filter classes (see #658).

### CODE REVIEW

File: tests/test_filter_gammatone.py [hunk 1]
```diff
@@ -122,11 +122,11 @@ def test_gammatone_bands_reset_state():
...
     # check for equality
-    npt.assert_array_equal(real_a.time, real.time[:, :2**11])
-    npt.assert_array_equal(imag_a.time, imag.time[:, :2**11])
+    npt.assert_array_equal(real_a.time, real.time[:, :, :2**11])
+    npt.assert_array_equal(imag_a.time, imag.time[:, :, :2**11])
 
-    npt.assert_array_equal(real_b.time, real.time[:, -2**11:])
-    npt.assert_array_equal(imag_b.time, imag.time[:, -2**11:])
+    npt.assert_array_equal(real_b.time, real.time[:, :, -2**11:])
+    npt.assert_array_equal(imag_b.time, imag.time[:, :, -2**11:])
```
Comment (ahms5) [inline tests/test_filter_gammatone.py:129]: maybe also add a test for the shape

**Review by ahms5 — CHANGES_REQUESTED** [review]
Thank you, just a small comment

**Comment by hoyer-a** [comment]
I added tests for the shape, once #664 is merged i will include the `shape=(1, 1)` case.

File: pyfar/dsp/filter/gammatone.py [hunk 2]
```diff
@@ -309,7 +309,7 @@ def _get_delays_and_phase_factors(self):
             delay_samples + 3, sampling_rate=self.sampling_rate))
 
         # compute the envelope
-        ir = real.time + 1j * imag.time
+        ir = real.time[:, 0, :] + 1j * imag.time[:, 0, :]
```
Comment (ahms5) [inline pyfar/dsp/filter/gammatone.py:312]:
```suggestion
        ir = real.time[..., 0, :] + 1j * imag.time[..., 0, :]
```
I am not sure about this, it looks very dangerous. Is there always a new dimension at dim=-2?

**Review by ahms5 — CHANGES_REQUESTED** [review]
Thanks for taking care. just one open question.

File: pyfar/dsp/filter/gammatone.py (hunk 2 above)
Comment (hoyer-a) [reply pyfar/dsp/filter/gammatone.py:312]: It should be always at dim=-2.  The impulse being used in `_get_delays_and_phase_factors` only varies in length and sampling rate.

**Review by ahms5 — APPROVED** [review]

File: pyfar/dsp/filter/gammatone.py (hunk 2 above)
Comment (f-brinkmann) [reply pyfar/dsp/filter/gammatone.py:312]: Yes, I think it is fine this way

**Review by f-brinkmann — APPROVED** [review]
Thanks - Looks good to me.

**Pull request merged**